code/core/AccessHelper.cc

```diff
@@ -1,5 +1,26 @@
 #include "AccessHelper.h"
 
+// ===============
+// ----* Hit *----
+// ===============
+
+//____________________________________________________________________________________________
+std::tuple<std::vector<unsigned int>, std::vector<unsigned int>> convertHitsToHitIdxsAndHitTypes(SDL::Event* event, std::vector<unsigned int> hits)
```

Comment:
Are these functions intended to run also as part of standalone code that produces an output to be used for CMSSW or they are just there for the purpose of the NTuple? If the former is the case, then we should consider adding them to the CMSSW class, with the proper adjustments to reduce host-device memory copies. Maybe it's too early to know that but, in any case, it's good to keep in mind that something additional may be needed here in the future.

Reply:
This is only for ntupling. afaik none of the functions in AccessHelper are not used for main results. (are they..?)

Reply:
They are. If we want to get the output of hit indices, pt, eta, etc. for usage in CMSSW, a modified version of some of them is used. The modifications were necessary to reduce device-host memory transfers. You can take a look at [SDL::LST::getOutput](https://github.com/SegmentLinking/TrackLooper/blob/c4551a751b9fef5d6261ff6974a5910a5ace0e7c/SDL/LST.cc#L369) and [Object accessors](https://github.com/SegmentLinking/TrackLooper/blob/c4551a751b9fef5d6261ff6974a5910a5ace0e7c/SDL/LST.cc#L525).

But, in any case, if this meant on for NTupling for now, it is ok and we can revisit it in the future.